src/internal/hooks/use-intersection-observer/index.ts

```diff
@@ -3,13 +3,19 @@
 
 import { RefCallback, useCallback, useRef, useState } from 'react';
 
+interface UseIntersectionObserverConfig {
+  initialState?: boolean;
+}
+
 /**
  * A hook that uses an Intersection Observer on the target element ref
  * and detects if the element is intersecting with its parent.
  */
-export function useIntersectionObserver<T extends HTMLElement>() {
+export function useIntersectionObserver<T extends HTMLElement>({
+  initialState = false,
```

Comment:
Made this configurable, because there are different use-cases

1. For lazy-loading something you want the initial value to be `false`
2. For suspending some operations when hiding you want the initial value to be `true`